### DISCUSSION TITLE
Deferring effect triggering

### DISCUSSION THREAD

**Discussion by predaytor** [preactjs/signals#602] [Ideas]
Solid.js has a [`on`](https://docs.solidjs.com/reference/reactive-utilities/on) utility that allows you to explicitly define dependencies with the option to defer until the next value changes. Maybe we have something similar for @preact/signals?

```tsx
const [a, setA] = createSignal();

// doesn't run immediately
createEffect(on(a, (v) => console.log(v), { defer: true }));

setA("new"); // now it runs
```

**Comment by rschristian** [comment]
You can pretty easily support this with a simple `initialized` value:

```js
import { signal, effect, useSignalEffect } from '@preact/signals';

const count = signal(0);
const initialized = signal(false);

effect(() => {
	count.value;
	if (!initialized.peek()) return initialized.value = true;
	console.log(count.value);
});

count.value++;
```

This does get more complicated the more an effect does, but we'd encourage users to keep effects small.

Edit: `initialized` as a signal might be overkill, depends on what you're doing. Standard variable (`let initialized = false`) can be fine too.

**Reply by predaytor** [reply]
thx!
